core/common/src/test/java/alluxio/util/io/FileUtilsTest.java

```diff
@@ -256,4 +263,67 @@ public void createDirTest() throws IOException {
     Assert.assertTrue(FileUtils.exists(tempDir.getAbsolutePath()));
     Assert.assertTrue(tempDir.delete());
   }
+
+  /**
+   * Tests {@link FileUtils#createBlockPath} method when storage dir exists or doesn't exist.
+   */
+  @Test
+  public void createStorageDirPathTest() throws IOException {
+    File storageDir = new File(mTestFolder.getRoot(), "storageDir");
+    File blockFile = new File(storageDir, "200");
+
+    // When storage dir doesn't exist
+    FileUtils.createBlockPath(blockFile.getAbsolutePath());
+    Assert.assertTrue(FileUtils.exists(storageDir.getAbsolutePath()));
+    Assert.assertEquals(
+        PosixFilePermissions.fromString("rwxrwxrwx"),
+        Files.getPosixFilePermissions(Paths.get(storageDir.getAbsolutePath())));
+
+    // When storage dir exists
+    FileUtils.createBlockPath(blockFile.getAbsolutePath());
+    Assert.assertTrue(FileUtils.exists(storageDir.getAbsolutePath()));
+  }
+
+  /**
+   * Tests invoking {@link FileUtils#createBlockPath} method concurrently. This simulates the case
+   * when multiple blocks belonging to the same storage dir get created concurrently.
+   */
+  @Test
+  public void concurrentCreateStorageDirPathTest() throws Exception {
```

Comment:
Does it fail now without the change? It would only take 2 lines to add CyclicBarrier to improve concurrency,

```java
final CyclicBarrier barrier = new CyclicBarrier(numCreators);
```
at the start, and
```java
barrier.await();
```
right before calling createBlockPath.

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CyclicBarrier.html